### PR TITLE
[FLASK] Add snap update metrics

### LINKED ISSUE #15202: Utilize `snapUpdated` event in metrics
Currently, we're only tracking snap installs. We should also be tracking snap updates.

### PR DESCRIPTION
**Pull request by hmalik88**
Closes #15202 

We were not utilizing the `snapUpdated` event to track snap updates, this PR adds code to do so.

### CODE REVIEW

File: app/scripts/metamask-controller.js [hunk 1]
```diff
@@ -1283,6 +1283,21 @@ export default class MetamaskController extends EventEmitter {
       },
     );
 
+    this.controllerMessenger.subscribe(
+      `${this.snapController.name}:snapUpdated`,
```
Comment (FrederikBolding) [inline app/scripts/metamask-controller.js:1287]: This is not compatible with the current version of skunkworks and needs to wait until we cut that release and include it in the extension.

**Review by FrederikBolding — CHANGES_REQUESTED** [review]
See above

File: app/scripts/metamask-controller.js (hunk 1 above)
Comment (hmalik88) [reply app/scripts/metamask-controller.js:1287]: ok, will keep this on standby.

**Review by FrederikBolding — DISMISSED** [review]

File: app/scripts/metamask-controller.js [hunk 2]
```diff
@@ -1291,6 +1291,21 @@ export default class MetamaskController extends EventEmitter {
...
+      (newSnap, oldVersion) => {
+        this.metaMetricsController.trackEvent({
+          event: 'Snap Updated',
+          category: EVENT.CATEGORIES.SNAPS,
+          properties: {
+            snap_id: newSnap.id,
+            oldVersion,
+            newVersion: newSnap.version,
```
Comment (brad-decker) [inline app/scripts/metamask-controller.js:1303]: All properties must be snake cased. Is this event tracked in John Brennan's schema?

**Review by brad-decker — DISMISSED** [review]

File: app/scripts/metamask-controller.js [hunk 3]
```diff
@@ -1291,6 +1291,21 @@ export default class MetamaskController extends EventEmitter {
       },
     );
 
+    this.controllerMessenger.subscribe(
+      `${this.snapController.name}:snapUpdated`,
+      (newSnap, oldVersion) => {
```
Comment (digiwand) [inline app/scripts/metamask-controller.js:1296]: [optional] it might be helpful to rename `newSnap` to `newTruncatedSnap` to preserve the type in the name

**Review by digiwand — APPROVED** [review]
👍

**Review by FrederikBolding — APPROVED** [review]

**Pull request merged**